app/jobs/characterize_job.rb

```diff
@@ -3,10 +3,14 @@ class CharacterizeJob < Hyrax::ApplicationJob
 
   # Characterizes the file at 'filepath' if available, otherwise, pulls a copy from the repository
   # and runs characterization on that file.
-  # @param [FileSet] file_set
+  # @param [FileSet, Valkyrie::Resource] file_set
   # @param [String] file_id identifier for a Hydra::PCDM::File
   # @param [String, NilClass] filepath the cached file within the Hyrax.config.working_path
   def perform(file_set, file_id, filepath = nil)
```

Comment:
It might be nicer to existing applications to follow the pattern used in `VisibilityCopyJob`:

  - extract the existing `#perform` implementation to a service, making the Job class single responsibility;
  - call the service from a factory like `Hyrax::Charactarizer.for(file_set: file_set).characterize`;
  - add a new implementation, reachable from the factory, for the valkyrie file_set.

This keeps the existing implementation from changing (apart from the extraction), and isolates branching logic to the factory.

Reply:
I am not trying to valkyrize this class at this time.  I am trying to valkyrize another class and just want this one not to barf when it is passed a valkyrie resource instead of a fileset.  This is reflected in the STATUS and TODO comments.

Reply:
if we're not wanting to change this job's behavior, why not do the conversion in the caller? That avoids expanding the method signature.

The Job is also already deserializing from a `GlobalId`. It ought to be possible to queue the job from just an ID and deserialize directly from `ActiveFedora` with no changes here.

Reply:
I am admittedly trying to take the most expedient approach with minimal change. The caller will be valkyrized and I do not want to touch the caller again for the valkyrization transition.  Whereas this class has not yet been evaluated as to what is best to do for the valkyrization transition.  Making this very minor change here allows the other work to be completed.  It is assumed that this class will be evaluated in the future to determine if additional changes are required.